File: server/src/main/java/io/github/hulang1024/chess/games/chess/ChessHost.java
package io.github.hulang1024.chess.games.chess;

/**
 * 棋方
 */
public enum ChessHost {
    FIRST(1),
    SECOND(2);

    private int code;

    ChessHost(int code) {
        this.code = code;
    }

    public int code() {
        return this.code;
    }

    public ChessHost reverse() {
        return this == FIRST ? SECOND : FIRST;
    }

    public static ChessHost from(int code) {
        switch (code) {
            case 1: return FIRST;
            case 2: return SECOND;
            default: return null;
        }
    }
}

File: client/src/rulesets/chess_host.ts
/**
 * 棋方
 */
enum ChessHost {
    /**
     * 先手
     */
    FIRST = 1,

    /**
     * 后手
     */
    SECOND = 2
}

export default ChessHost;

// eslint-disable-next-line
namespace ChessHost {
  // eslint-disable-next-line
  export function reverse(host: ChessHost) {
    return host == ChessHost.SECOND ? ChessHost.FIRST : ChessHost.SECOND;
  }
}
